beacon-chain/blockchain/service.go

```diff
@@ -284,6 +284,18 @@ func (c *ChainService) blockProcessing(done <-chan struct{}) {
 					log.Errorf("could not process attestation for block %d because %v", block.SlotNumber(), err)
 				} else {
 					canProcessAttestations = true
+					// Save processed attestation to local db.
```

Comment:
A general note: this method is a mess and needs to be refactored. 
We shouldn't have so many layers of logic in a single method.

Reply:
Agreed. I believe #474 and #473 will address that

Reply:
Yeah, you can take a look at #473. the logic's a lot cleaner (i think). The code can become a lot simpler if we use Bolt where database updates can be abstracted with one method call.